File: src/main/java/org/polyfrost/glintcolorizer/config/category/BaseGlint.java
package org.polyfrost.glintcolorizer.config.category;

import org.polyfrost.oneconfig.api.config.v1.annotations.Color;
import org.polyfrost.oneconfig.api.config.v1.annotations.Slider;
import org.polyfrost.oneconfig.api.config.v1.annotations.Switch;
import org.polyfrost.polyui.color.ColorUtils;
import org.polyfrost.polyui.color.PolyColor;

public class BaseGlint {
	@Switch(title = "Enabled")
	public boolean enabled = true;

	@Switch(title = "Individual Strokes")
	public boolean individualStrokes = false;

	@Color(title = "Color")
	public PolyColor color = ColorUtils.rgba(0x80, 0x40, 0xCC, 0xFF);

	@Color(title = "Stroke One Color")
	public PolyColor strokeOneColor = ColorUtils.rgba(0xFF, 0x00, 0x00, 0xFF);

	@Slider(title = "Stroke Two Rotation", min = -50, max = 50, step = 1)
	public int strokeOneRotation = -50;

	@Color(title = "Stroke Two Color")
	public PolyColor strokeTwoColor = ColorUtils.rgba(0x0A, 0xEA, 0xFF, 0xFF);

	@Slider(title = "Stroke Two Rotation", min = -50, max = 50, step = 1)
	public int strokeTwoRotation = 10;

	@Slider(title = "Speed", max = 2.0F, step = 0.1F)
	public float speed = 1.0F;

	@Slider(title = "Scale", max = 2.0F)
	public float scale = 1.0F;
}

File: src/main/kotlin/org/polyfrost/glintcolorizer/config/BaseGlint.kt
package org.polyfrost.glintcolorizer.config

import org.polyfrost.oneconfig.api.config.v1.annotations.Color
import org.polyfrost.oneconfig.api.config.v1.annotations.Slider
import org.polyfrost.oneconfig.api.config.v1.annotations.Switch
import org.polyfrost.polyui.color.argb

open class BaseGlint {
    @Switch(title = "Enable Glint Visual")
    var enabled = true

    @Switch(title = "Modify Strokes Individually")
    var individualStrokes = true

    @Color(title = "Glint Color", description = "Modifies the color of the enchantment glint.")
    var glintColor = argb(GlintConfig.DEFAULT_GLINT_COLOR)

    @Color(title = "Stroke 1 Color", description = "Modifies the first stroke of the enchantment glint effect.")
    var strokeOneColor = argb(0xFFFF0000.toInt()) // argb(GlintConfig.DEFAULT_GLINT_COLOR)

    @Slider(title = "Stroke 1 Rotation", min = -180.0F, max = 180.0F)
    var firstStrokeRotation = -50.0F

    @Color(title = "Stroke 2 Color", description = "Modifies the second stroke of the enchantment glint effect.")
    var strokeTwoColor = argb(0xFF0AEAFF.toInt()) // argb(GlintConfig.DEFAULT_GLINT_COLOR)

    @Slider(title = "Stroke 2 Rotation", min = -180.0F, max = 180.0F)
    var secondStrokeRotation = 10.0F

    @Slider(title = "Speed", min = 0.1F, max = 10.0F)
    var speed = 1.0F

    @Slider(title = "Scale", min = 0.0F, max = 8.0F)
    var scale = 1.0F
}